app/components/primer/dropdown.rb

```diff
@@ -0,0 +1,127 @@
+    # Required trigger for the dropdown. Has the same arguments as <%= link_to_component(Primer::ButtonComponent) %>,
+    # but it is locked as a `summary` tag.
+    renders_one :summary, lambda { |**system_arguments, &block|
+      @summary_arguments = system_arguments
+      @summary_arguments[:button] = true
+
+      view_context.capture { block&.call }
+    }
+
+    # Required context menu for the dropdown
+    #
+    # @param as [Symbol] When `as` is `:list`, wraps the menu in a `<ul>` with a `<li>` for each item.
+    # @param direction [Symbol] <%= one_of(Primer::Dropdown::Menu::DIRECTION_OPTIONS) %>
+    # @param scheme [Symbol] Pass `:dark` for dark mode theming
+    # @param header [String] Optional string to display as the header
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    renders_one :menu, "Primer::Dropdown::Menu"
+
+    # @example Default
+    #   <div>
+    #     <%= render(Primer::Dropdown.new) do |c| %>
+    #       <% c.summary do %>
+    #         Dropdown
+    #       <% end %>
+    #
+    #       <%= c.menu(header: "Options") do |menu|
+    #         menu.item { "Item 1" }
+    #         menu.item { "Item 2" }
+    #         menu.item(divider: true)
+    #         menu.item { "Item 3" }
+    #         menu.item { "Item 4" }
+    #       end %>
+    #     <% end %>
+    #   </div>
+    #
+    # @example With Direction
+    #   <div>
+    #     <%= render(Primer::Dropdown.new(display: :inline_block)) do |c| %>
+    #       <% c.summary do %>
+    #         Dropdown
+    #       <% end %>
+    #
+    #       <%= c.menu(header: "Options", direction: :s) do |menu|
+    #         menu.item { "Item 1" }
+    #         menu.item { "Item 2" }
+    #         menu.item(divider: true)
+    #         menu.item { "Item 3" }
+    #         menu.item { "Item 4" }
+    #       end %>
+    #     <% end %>
+    #   </div>
+    #
+    # @example Customizing the button
+    #   <div>
+    #     <%= render(Primer::Dropdown.new) do |c| %>
+    #       <% c.summary(scheme: :primary, variant: :small) do %>
+    #         Dropdown
+    #       <% end %>
+    #
+    #       <%= c.menu(header: "Options") do |menu|
+    #         menu.item { "Item 1" }
+    #         menu.item { "Item 2" }
+    #         menu.item(divider: true)
+    #         menu.item { "Item 3" }
+    #         menu.item { "Item 4" }
+    #       end %>
+    #     <% end %>
+    #   </div>
+    #
+    # @example Menu as list
+    #   <div>
+    #     <%= render(Primer::Dropdown.new) do |c| %>
+    #       <% c.summary do %>
+    #         Dropdown
+    #       <% end %>
+    #
+    #       <%= c.menu(as: :list, header: "Options") do |menu|
+    #         menu.item { "Item 1" }
+    #         menu.item { "Item 2" }
+    #         menu.item(divider: true)
+    #         menu.item { "Item 3" }
+    #         menu.item { "Item 4" }
+    #       end %>
+    #     <% end %>
+    #   </div>
+    #
+    # @example Customizing menu items
+    #   <div>
```

Comment:
Is there any way to avoid these wrapping `div`s in these examples?

Reply:
I'm pretty sure we need them for the dropdown to render correctly, but I'll double check